spharpy/plot/spatial.py

```diff
@@ -952,8 +960,11 @@ def contour_map(
 
     ax.contour(xi, yi, zi, levels=levels, linewidths=0.5, colors='k',
                vmin=limits[0], vmax=limits[1], extend=extend)
-    cf = ax.pcolormesh(xi, yi, zi, cmap=cmap, shading='gouraud',
-                       vmin=limits[0], vmax=limits[1])
+    cf = ax.contourf(xi, yi, zi, levels=levels, cmap=cmap, vmin=limits[0],
+                     vmax=limits[1])
```

Comment:
Actually, these can now be replaced using the private function _combined_contour, which removes the need for interpolating the data. This makes the function a lot slimmer already.